internal/httputil/mux.go

```diff
@@ -9,34 +11,55 @@ import (
 	"github.com/TykTechnologies/tyk/internal/maps"
 )
 
-// routeCache holds the raw routes as they are mapped to mux regular expressions.
-// e.g. `/foo` becomes `^/foo$` or similar, and parameters get matched and replaced.
+// routeCache holds the raw routes as they are mapped from mux parameters to regular expressions.
+// e.g. `/foo/{id}` becomes `^/foo/([^/]+)$` or similar.
 var pathRegexpCache = maps.NewStringMap()
 
-// GetPathRegexp will convert a mux route url to a regular expression string.
-// The results for subsequent invocations with the same parameters are cached.
-func GetPathRegexp(pattern string) (string, error) {
-	val, ok := pathRegexpCache.Get(pattern)
+// apiLandIDsRegex matches mux-style parameters like `{id}`.
+var apiLangIDsRegex = regexp.MustCompile(`{([^}]+)}`)
```

Comment:
```suggestion
var apiLangIDsRegex = regexp.MustCompile(`{([^}]+)}`)
```

I've seen listenpath hacks that provide `/{}/users` as the URL. Wondering if this should remain supported.